### ISSUE TITLE
Disorder result when I try 投

### ISSUE THREAD

**Issue by moonfighting** [tensorflow/models#1129]
I have used[]( https://github.com/tensorflow/models/tree/master/slim/train_image_classifier.py) to finetune from the pretrained inception-resnet-v2 model with my own dataset and get several new models.

Now I want to have a test on these models. In order to get every image's score of all the classes, I write my  evaluation function, the code is shown below:

```
def evaluate(master, checkpoint_path, 
                  logdir, num_evals = 1, 
                  prob_op = None, 
                 eval_op_feed_dict = None, 
                  labels = None, 
                  variables_to_restore = None, 
                  session_config = None):
     
      global_step = variables.get_or_create_global_step()
      saver = tf_saver.Saver(variables_to_restore or variables.get_variables_to_restore())
      sv = supervisor.Supervisor(graph = ops.get_default_graph(),
                                  logdir = logdir,
                                  summary_op = None,
                                  summary_writer = None,
                                  global_step = None,
                                  saver = None)
      with sv.managed_session(master,
                             start_standard_services = False,
                              config = session_config) as sess:
          saver.restore(sess, checkpoint_path)
          sv.start_queue_runners(sess)
          if prob_op is not None:
              for i in range(int(num_evals)):
                 print('probs:', sess.run(prob_op))
                  if labels is not None:
                      print('labels:', sess.run(labels))

```
However, the result is disordered,  there are some repetitive results and the label is not corresponding to the image. 
An example of the disordered result is shown below:
```
probs: [[ 0.00452999  0.98399526  0.0114748 ]]
labels: [[1]]
probs: [[  5.94390403e-05   9.17715430e-01   8.22251886e-02]]
labels: [[0]]
probs: [[  9.99693394e-01   2.97987135e-04   8.66993832e-06]]
labels: [[0]]
probs: [[  1.88809641e-14   2.11972529e-13   1.00000000e+00]]
labels: [[2]]
probs: [[ 0.00116475  0.11707634  0.88175893]]
labels: [[2]]
probs: [[ 0.00452999  0.98399526  0.0114748 ]]
labels: [[1]]
probs: [[  3.82530561e-04   1.43166608e-03   9.98185813e-01]]
labels: [[0]]
probs: [[  9.77581263e-01   2.22077761e-02   2.10878396e-04]]
labels: [[0]]
probs: [[  1.88809641e-14   2.11972529e-13   1.00000000e+00]]
labels: [[2]]
probs: [[  7.67493984e-05   1.54769485e-04   9.99768436e-01]]
labels: [[2]]
```
I am sure that the ten images are all different(with different md5), but the first line in the result is equal to the 11th line, and the 7th line is also equal to 17th line. 

**the tf's version is 0.12**
**The result confused me, hoping someone could help me**

**Comment by prb12** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!
